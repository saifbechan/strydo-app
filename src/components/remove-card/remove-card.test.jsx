import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import mockStore from './../../redux/store.mock';

import RemoveCard, {
  RemoveCard as RemoveCardPure
} from './remove-card.component';

const props = {
  columnId: 'abc-123',
  cardId: 'def-456'
};

it('renders correctly', () => {
  const { asFragment } = render(<RemoveCardPure {...props} />);
  expect(asFragment()).toMatchSnapshot();
});

it('calls removeCard with the correct payload on click', () => {
  const store = mockStore();

  const { getByTestId } = render(<RemoveCard store={store} {...props} />);

  fireEvent.click(getByTestId('remove-card-handler'));

  expect(store.getActions()).toEqual([
    { type: 'REMOVE_CARD', payload: { ...props } }
  ]);
});
