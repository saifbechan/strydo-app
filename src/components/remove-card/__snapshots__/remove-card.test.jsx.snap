// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  cursor: pointer;
  margin-top: 15px;
  padding: 5px;
  text-align: center;
  width: 100%;
}

.c0:hover {
  color: grey;
}

<button
    class="c0"
    data-testid="remove-card-handler"
  >
    - remove card
  </button>
</DocumentFragment>
`;
